assets/js/modules/analytics/components/module/ModuleOverviewWidget/SiteStats.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * SiteStats component the ModuleOverviewWidget widget.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import { extractAnalyticsDashboardData, getTimeColumnVaxisFormat } from '../../../util';
+import GoogleChart from '../../../../../components/GoogleChart';
+import PreviewBlock from '../../../../../components/PreviewBlock';
+import { Cell, Row, Grid } from '../../../../../material-components';
+const { useSelect } = Data;
+
+export default function SiteStats( { loaded, selectedStat, report } ) {
+	const currentDayCount = useSelect( ( select ) => select( CORE_USER ).getDateRangeNumberOfDays() );
+
+	if ( ! loaded ) {
+		return (
+			<PreviewBlock width="100%" height="270px" />
+		);
+	}
+
+	const dataMap = extractAnalyticsDashboardData( report, selectedStat, currentDayCount, 0, 1 );
```

Comment:
@eugene-manuilov Following up on our previous conversation: My main issue with this function is that it is for a super-specific purpose; I'd prefer if it wasn't a general utility function but rather lived close to the one or two components that it is specific to.

We can keep this for now, but I think we should revisit it once we remove the legacy site stats component. Could be as simple as moving right into this file and not exporting it.

Reply:
Yes, agreed. Once we remove the legacy component, we won't need it to be in the utils folder and can simply move it to the component file.